server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -74,15 +74,16 @@ def validate
     end
 
     def log
-      reader = Fusor::Logging::LogReader.new
-      log_path = ::Fusor.log_file_path(@deployment.name, @deployment.id)
+      log_type_param = params[:log_type] || 'fusor_log'
+      reader = create_log_reader(log_type_param)
```

Comment:
I think this might work:

```
class MockReader
end

mock_reader = MockReader.new
mock_reader.expects(:tail_log_since).once.returns(...)
ApplicationController.any_instance.stubs(: create_log_reader).returns(mock_reader)
```